### PR TITLE
chore(npm/angular): migrate angular adapter to monorepo

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes N/A

### User facing changelog

Migrate Angular adapter into monorepo from https://www.npmjs.com/package/cypress-angular-unit-test.

Still WIP. It's marked `private: true` so it won't go to npm yet.

### Additional details

Consolidate all major adapters (Vue, React, Angular) in a single place This supports better code sharing, and the ability to run their test suite against the latest Cypress commits.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by lmiller1990** [comment]
Trying to test but cannot create a new angular app using the angular-cli. Weird.

File: npm/angular/package.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    "@cypress/code-coverage": "3.9.5",
+    "@cypress/webpack-dev-server": "1.1.4",
+    "@cypress/webpack-preprocessor": "5.7.0",
+    "@types/cypress-image-snapshot": "3.1.5",
+    "@types/node": "8.10.66",
+    "angular-router-loader": "0.8.5",
+    "angular2-template-loader": "0.6.2",
+    "codelyzer": "6.0.2",
+    "core-js": "3.12.1",
+    "cypress": "7.1.0",
```
Comment (agg23) [inline npm/angular/package.json:48]: Shouldn't this be local Cypress?

**Review by agg23 — DISMISSED** [review]

File: npm/angular/package.json (hunk 1 above)
Comment (lmiller1990) [reply npm/angular/package.json:48]: Yes
the test script uses local but I will remove this dependency

Comment (lmiller1990) [reply npm/angular/package.json]: Done.

**Review by agg23 — DISMISSED** [review]

File: npm/angular/.github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (elevatebart) [inline npm/angular/.github/ISSUE_TEMPLATE/bug_report.md:1]: I believ you can remove those

File: npm/angular/.prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (elevatebart) [inline npm/angular/.prettierrc.json:1]: Ooooh this is going to be controversial !!

File: npm/angular/renovate.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (elevatebart) [inline npm/angular/renovate.json:1]: remove renovate from here ;)

**Review by elevatebart — COMMENTED** [review]
Great PR! I still have a couple of comments.

- I believe the snapshot images taken by cypress should not be version controlled.
- npm/angular/src/assets/.gitkeep is unnecessary since we have a logo for cypress besides it

File: npm/angular/.prettierrc.json (hunk 3 above)
Comment (lmiller1990) [reply npm/angular/.prettierrc.json:1]: Removed, let's avoid controversy for now :)

**Comment by lmiller1990** [comment]
@elevatebart fixed most of those things. Why *don't* we add snapshots? I think we need them, so we can diff the new ones with the old ones. This repo uses 'cypress-image-snapshot`, which does the diff between existing ones and new ones - like visual regression.

**Comment by elevatebart** [comment]
The design-system test are failing because of a classic cypress flake (the element is detached from the DOM)

The tests are fixed in [this PR](https://github.com/cypress-io/cypress/pull/16502) that still needs to be approved/merged. 

I believe you will have to do so before this one can pass.

File: npm/angular/src/app/network/network.component.cy-spec.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { HttpClientModule } from '@angular/common/http'
+import { initEnv, mount } from '@cypress/angular'
+import { NetworkService } from '../network.service'
+import { NetworkComponent } from './network.component'
+
+describe('Network', () => {
```
Comment (LeJeanbono) [inline npm/angular/src/app/network/network.component.cy-spec.ts:6]: I just added an example with `cy.intercept` : 
```
import { HttpClientModule } from '@angular/common/http';
import { initEnv, mount } from 'cypress-angular-unit-test';
import { NetworkService } from '../network.service';
import { NetworkComponent } from './network.component';

describe('Network', () => {
  describe('No mock', () => {
    beforeEach(() => {
      initEnv(NetworkComponent, {
        providers: [NetworkService],
        imports: [HttpClientModule],
      });
    });

    it('fetches 3 users from remote API', () => {
      mount(NetworkComponent);
      cy.get('li').should('have.length', 3);
    });
  });
  describe('With mock', () => {
    beforeEach(() => {
      cy.intercept('/users?_limit=3', { body: [{ id: 1, name: 'foo' }] });
      initEnv(NetworkComponent, {
        providers: [NetworkService],
        imports: [HttpClientModule],
      });
    });
    it('fetch 1 user stubed', () => {
      mount(NetworkComponent);
      cy.get('li').should('have.length', 1).first().contains('foo');
    });
  });
});

```

Comment (lmiller1990) [reply npm/angular/src/app/network/network.component.cy-spec.ts:6]: I'll add this, thanks.

Feel free to make PRs against this repo if you like 👍 or just ping me and I can make changes (still learning Angular...)

Comment (lmiller1990) [reply npm/angular/src/app/network/network.component.cy-spec.ts:6]: Done

**Review by elevatebart — APPROVED** [review]
GOood stuff

**Pull request merged**